cassandraprotocol/frame/decoder.go

```diff
@@ -19,19 +19,30 @@ func (c *Codec) Decode(source io.Reader) (*Frame, error) {
 	} else {
 		header := &Header{
 			Version:          rawHeader.Version,
-			StreamId:         int16(rawHeader.StreamId & 0xFFFF),
+			StreamId:         rawHeader.StreamId,
 			TracingRequested: body.TracingId != nil || rawHeader.Flags&cassandraprotocol.HeaderFlagTracing != 0,
 		}
 		return &Frame{Header: header, Body: body}, nil
 	}
 }
 
+// Decode decodes the entre frame, decompressing the body if needed.
+func (c *Codec) DecodeRaw(source io.Reader) (*RawFrame, error) {
+	if rawHeader, err := c.DecodeHeader(source); err != nil {
+		return nil, err
```

Comment:
So far I've been using the convention of enriching the error messages as they bubble up the chain. Here, I would rather wrap the error returned by `DecodeHeader` in another error:
```suggestion
		return nil, fmt.Errorf("cannot decode header: %w", err)
```
The same would apply to the call to `ReadBody` below.

Reply:
https://github.com/datastax/go-cassandra-native-protocol/pull/5#discussion_r512689303